tools/gatk4.wdl

```diff
@@ -0,0 +1,333 @@
+        File dict
+        String prefix = basename(bam, ".bam") + ".split"
+        Int modify_disk_size_gb = 0
+    }
+
+    Int disk_size_gb = ceil(size(bam, "GB") + 1) * 5 + ceil(size(fasta, "GB")) + modify_disk_size_gb
+
+    command <<<
+        set -euo pipefail
+
+        gatk \
+            SplitNCigarReads \
+            -R ~{fasta} \
+            -I ~{bam} \
+            -O ~{prefix}.bam \
+            -OBM true
+       # GATK is unreasonable and uses the plain ".bai" suffix.
+       mv ~{prefix}.bai ~{prefix}.bam.bai
+    >>>
+
+    output {
+        File split_bam = "~{prefix}.bam"
+        File split_bam_index = "~{prefix}.bam.bai"
+        File split_bam_md5 = "~{prefix}.bam.md5"
+    }
+
+    runtime {
+        cpu: 8
+        memory: "24 GB"
+        disk: "~{disk_size_gb} GB"
+        container: "quay.io/biocontainers/gatk4:4.4.0.0--py36hdfd78af_0"
+        maxRetries: 1
+    }
+}
+
+task base_recalibrator {
+    meta {
+        description: "Generates recalibration report for base quality score recalibration."
+        external_help: "https://gatk.broadinstitute.org/hc/en-us/articles/360036897372-BaseRecalibratorSpark-BETA"
+        outputs: {
+            recalibration_report: "Recalibration report file"
+        }
+
+    }
+
+    parameter_meta  {
+        bam: "Input BAM format file on which to recabilbrate base quality scores"
+        bam_index: "BAM index file corresponding to the input BAM"
+        fasta: "Reference genome in FASTA format"
+        fasta_index: "Index for FASTA format genome"
+        dict: "Dictionary file for FASTA format genome"
+        dbSNP_vcf: "dbSNP VCF file"
+        dbSNP_vcf_index: "dbSNP VCF index file"
+        known_indels_sites_VCFs: "List of VCF files containing known indels"
+        known_indels_sites_indices: "List of VCF index files corresponding to the VCF files in `known_indels_sites_VCFs`"
+        use_original_quality_scores: "Use original quality scores from the input BAM. Default is to use recalibrated quality scores."
+        prefix: "Prefix for the output recalibration report. The extension `.txt` will be added."
+        modify_memory_gb: "Add to or subtract from dynamic memory allocation. Default memory is determined by the size of the inputs. Specified in GB."
+    }
+
+    input {
+        File bam
+        File bam_index
+        String prefix = basename(bam, ".bam") + ".recal"
+        File dict
+        File fasta
+        File fasta_index
+        File dbSNP_vcf
+        File dbSNP_vcf_index
+        Array[File] known_indels_sites_VCFs
+        Array[File] known_indels_sites_indices
+        Int modify_disk_size_gb = 0
+        Boolean use_original_quality_scores = true
+    }
+
+    Int disk_size_gb = ceil(size(bam, "GB") + 1) * 3 + ceil(size(fasta, "GB")) + modify_disk_size_gb
+
+    command <<<
+        gatk \
+            --java-options "-XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10 -XX:+PrintFlagsFinal \
+            -Xloggc:gc_log.log -Xms4000m" \
```

Comment:
`GCHeapFreeLimit` is the amount of heap space that must be freed in a GC run (defaults to 2%). 
`GCTimeLimit` is the amount of time spent garbage collecting as a percentage of the total time (defaults to 98%). I suspect that these were introduced by the Broad to avoid cases where the process gets swamped by excessive GC runs that don't actually find much space to free. In those cases, it's probably better to throw an error and have the user resubmit with additional memory.